### PR TITLE
Update nanosam to take in bounding boxes instead of points

### PR DESCRIPTION
**Pull request by silkyar**
`nanosam` doesn't take in bounding boxes as input, and instead expect points. We could provide a bounding box's corners as points but this doesn't work with multiple bounding boxes. 
This PR makes some changes to add in `boxes` as an input param. It also removes the input params that we don't use (point, point_labels, masks) to ensure that the resulting TRT engine doesn't expect them.

### CODE REVIEW

File: nanosam/mobile_sam/utils/onnx.py
Comment (yaswanth-fauna) [inline nanosam/mobile_sam/utils/onnx.py:1]: May be we can remove the `_embed_points` function. Now that we replaced with  `embed_boxes`.

File: nanosam/mobile_sam/utils/onnx.py [hunk 1]
```diff
@@ -108,12 +139,15 @@ def select_masks(
...
+        # point_coords: torch.Tensor,
+        # point_labels: torch.Tensor,
+        # mask_input: torch.Tensor,
+        # has_mask_input: torch.Tensor,
+        bbox: torch.Tensor,
     ):
-        sparse_embedding = self._embed_points(point_coords, point_labels)
+        mask_input = torch.zeros(1, 1, 256, 256).float()
+        has_mask_input = torch.Tensor([0])
+        sparse_embedding = self._embed_boxes(input_image_size=(1024, 1024), boxes=bbox)
```
Comment (yaswanth-fauna) [inline nanosam/mobile_sam/utils/onnx.py:150]: I think we can get the image size (1024, 1024) from the prompt_encoder  `self.model.prompt_encoder.input_image_size`

**Review by yaswanth-fauna — APPROVED** [review]
Looks Great!! Added just few comments

File: nanosam/mobile_sam/utils/onnx.py
Comment (silkyar) [reply nanosam/mobile_sam/utils/onnx.py:1]: I am trying to not touch the code that I don't need. That is, I am only removing code that messes with generating the engine. Trying to reduce future pain if we need to use points.
